Mathlib/CategoryTheory/Sites/ChosenFiniteProducts.lean

```diff
@@ -31,20 +31,30 @@ variable (X Y : Sheaf J A)
 
 lemma tensorProd_isSheaf : Presheaf.IsSheaf J (X.val ⊗ Y.val) := by
   apply isSheaf_of_isLimit (E := (Cones.postcompose (pairComp X Y (sheafToPresheaf J A)).inv).obj
-    (ChosenFiniteProducts.product X.val Y.val).cone)
-  exact (IsLimit.postcomposeInvEquiv _ _).invFun (ChosenFiniteProducts.product X.val Y.val).isLimit
+    (BinaryFan.mk (ChosenFiniteProducts.fst X.val Y.val) (ChosenFiniteProducts.snd _ _)))
+  exact (IsLimit.postcomposeInvEquiv _ _).invFun
+    (ChosenFiniteProducts.tensorProductIsBinaryProduct X.val Y.val)
 
 lemma tensorUnit_isSheaf : Presheaf.IsSheaf J (𝟙_ (Cᵒᵖ ⥤ A)) := by
   apply isSheaf_of_isLimit (E := (Cones.postcompose (Functor.uniqueFromEmpty _).inv).obj
-    ChosenFiniteProducts.terminal.cone)
-  · exact (IsLimit.postcomposeInvEquiv _ _).invFun ChosenFiniteProducts.terminal.isLimit
+    (asEmptyCone (𝟙_ _)))
+  · exact (IsLimit.postcomposeInvEquiv _ _).invFun ChosenFiniteProducts.isTerminalTensorUnit
   · exact Functor.empty _
 
-/-- Any `ChosenFiniteProducts` on `A` induce a `ChosenFiniteProducts` structures on `A`-valued
-sheaves. -/
-@[simps! product_cone_pt_val terminal_cone_pt_val_obj terminal_cone_pt_val_map]
-noncomputable instance chosenFiniteProducts : ChosenFiniteProducts (Sheaf J A) where
-  product X Y :=
+/-- Any `ChosenFiniteProducts` on `A` induce a
+`ChosenFiniteProducts` structure on `A`-valued sheaves. -/
+noncomputable instance chosenFiniteProducts : ChosenFiniteProducts (Sheaf J A) :=
+  .ofChosenFiniteProducts
+   ({ cone := asEmptyCone { val := 𝟙_ (Cᵒᵖ ⥤ A)
+                            cond := tensorUnit_isSheaf _}
+      isLimit :=
+        { lift := fun f ↦ ⟨ChosenFiniteProducts.toUnit f.pt.val⟩
```

Comment:
```suggestion
        { lift f := ⟨ChosenFiniteProducts.toUnit f.pt.val⟩
```

Reply:
I'm not sure if I should be changing them if I am just moving things around (same for the one below). I'll incorporate the changes if you think I do.

Reply:
You've done unrelated cleanups already, so it's no worse IMO

Reply:
I don't think I have? But sure.